lib/ansible/modules/system/xfs_quota.py

```diff
@@ -0,0 +1,367 @@
+            module.fail_json(
+                msg='%s is not mounted with the uquota/usrquota/quota/uqnoenforce/qnoenforce option.'
+                    % mountpoint
+            )
+        try:
+            pwd.getpwnam(name)
+        except KeyError as e:
+            module.fail_json(msg='User %s doesn\'t exist.' % name)
+
+    if quota_type == 'group':
+        type_arg = '-g'
+        quota_default = 'root'
+        if name is None:
+            name = quota_default
+
+        if 'gquota' not in mp['mntopts'] and 'grpquota' not in mp['mntopts'] and 'gqnoenforce' not in mp['mntopts']:
+            module.fail_json(
+                msg='%s is not mounted with the gquota/grpquota/gqnoenforce option. (current options: %s)'
+                    % (mountpoint, mp['mntopts'])
+            )
+        try:
+            grp.getgrnam(name)
+        except KeyError as e:
+            module.fail_json(msg='User %s doesn\'t exist.' % name)
+
+    elif quota_type == 'project':
+        type_arg = '-p'
+        quota_default = '#0'
+        if name is None:
+            name = quota_default
+
+        if 'pquota' not in mp['mntopts'] and 'prjquota' not in mp['mntopts'] and 'pqnoenforce' not in mp['mntopts']:
+            module.fail_json(msg='%s is not mounted with the pquota/prjquota/pqnoenforce option.' % mountpoint)
+
+        if name != quota_default and not os.path.isfile('/etc/projects'):
+            module.fail_json(msg='/etc/projects doesn\'t exist.')
+
+        if name != quota_default and not os.path.isfile('/etc/projid'):
+            module.fail_json(msg='/etc/projid doesn\'t exist.')
+
+        if name != quota_default and name is not None and get_project_id(name) is None:
+            module.fail_json(msg='%s hasn\'t been defined in /etc/projid.' % name)
+
+        prj_set = True
+        if name != quota_default:
+            cmd = 'project %s' % name
+            r = exec_quota(module, cmd, mountpoint)
+            if r['rc'] != 0:
+                module.fail_json(msg='Could not get project state.', cmd=cmd, retval=r)
+            else:
+                for line in r['stdout']:
+                    if '%s - project identifier is not set' in line:
+                        prj_set = False
+                        break
+
+        if not prj_set and not module.check_mode:
+            cmd = 'project -s'
+            r = exec_quota(module, cmd, mountpoint)
+            if r['rc'] != 0:
+                module.fail_json(msg='Could not get quota realtime block report.', cmd=cmd, retval=r)
+            else:
+                changed = True
+        elif not prj_set and module.check_mode:
+            changed = True
+
+    changed = False
+
+    # Set limits
+    if state == 'absent':
+        bhard = 0
+        bsoft = 0
+        ihard = 0
+        isoft = 0
+        rtbhard = 0
+        rtbsoft = 0
+
+    if bsoft is not None or bhard is not None:
+        current_bsoft, current_bhard = quota_report(module, mountpoint, name, quota_type, 'b')
```

Comment:
I think it would be nice if the module could return the information.
So if you would put this into the `result` dictionary, that's easy.

Reply:
Not as facts, just as part of the exit_json() call. I would create at the very start of the module a result dictionary:
```python
    result = dict(
        changed = False,
    )
```

And update this dictionary with additional information you want to return.

Reply:
done